### PR TITLE
Correct default value of alphas

### PR DESCRIPTION
**Pull request by mtanghu**
The PopulationSummaryResults class should to default to self.alpha when the alpha parameter is not passed. This is not the case currently since some methods use the statement 'alpha = self.alpha if alpha is None else alpha', however alpha's default value in the methods is .1, thus self.alpha will be ignored even if the client does not pass the alpha parameter.

Without the fix ate_interval(), marginal_ate_interval() and const_marginal_ate_interval() all ignore the alpha parameter.

### CODE REVIEW

**Comment by heimengqi** [comment]
Thanks @mtanghu ! Great catch! I will address that thoroughly in a separate PR.

**Comment by heimengqi** [comment]
@mtanghu We are happy to have more external contributors. I think your current fix have already partially addressed this issue, and we have the similar issue for `value`, `tol` and `decimals` as well, it would be great if you'd like to implement those changes as well.

The logic will be in the initializer we have some default `alpha`, `value`, `tol`, `decimals`  since call `PopulationSummaryResult` will output the summary table, and if users still want to call `PopulationSummaryResult.conf_int_mean()` for instance, by default it will use the same alpha set from initializer, but if they call `.conf_int_mean(alpha=bla)`, it will update the CI with new alpha. In this case, default for `alpha`, `value`, `tol`, `decimals` in each method has to be None.

**Comment by mtanghu** [comment]
@heimengqi I'd love to! I'll do that this weekend!

File: econml/inference/_inference.py
Comment (kbattocchi) [inline econml/inference/_inference.py:1313]: As @heimengqi pointed out, it would be great to also address `tol` while you're at it.

File: econml/inference/_inference.py
Comment (kbattocchi) [inline econml/inference/_inference.py:1245]: Could you also please add a test to test_inference.py that fails with the current implementation but will be fixed by your changes?  For example, I believe that a test that trains an estimator, calls `ate_interval` with two different values of alpha, and asserts that the interval from the smaller alpha is wider would do this.

**Review by kbattocchi — CHANGES_REQUESTED** [review]
Thanks, this mostly looks good, but I've added a few extra suggestions (one of which was also made by @heimengqi).

Once those are addressed I'd be happy to approve this.

**Comment by kbattocchi** [comment]
@mtanghu Are you still interested in following this through, or should someone on our team do the remaining work to get this PR cleaned up and then merged?

**Comment by mtanghu** [comment]
@kbattocchi  Oh yes sorry, I'm definitely interested and just caught up in other work. I should be able to get this PR cleaned up soon!

**Comment by mtanghu** [comment]
This should be ready to merge, lmk if there's anything you else you'd like me to help with

File: econml/inference/_inference.py
Comment (mtanghu) [reply econml/inference/_inference.py:1282]: Should be done with the recent commits

**Review by kbattocchi — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**
